### ISSUE TITLE
Error parsing reference go1.10 as build when following README instructions

### ISSUE THREAD

**Issue by benjaminapetersen** [operator-framework/helm-app-operator-kit#23]
In the [instructions block](https://github.com/operator-framework/helm-app-operator-kit#instructions):

```
$ docker build -t quay.io/<namespace>/<chart>-operator --build-arg HELM_CHART=/path/to/helm/chart --build-arg API_VERSION=<group/version> --build-arg KIND=<Kind> .
```
I ran as the following:

```
$ docker build -t quay.io/benjaminapetersen/myapp-operator --build-arg=examples/myapp/ --build-arg API_VERSION=app.benjaminapetersen.me/v1alpha1 --build-arg KIND=MyApp .
```
And received this error output:

```
Sending build context to Docker daemon 324.6 kB
Step 1/20 : FROM golang:1.10 as builder
Error parsing reference: "golang:1.10 as builder" is not a valid repository/tag: invalid reference format
```
I had `go1.8.3`, so updated to `go1.10` and ran again, which resulted in the same output.  

I put my (working) helm chart within the project under `examples/myapp` as this somewhat seems to follow the suggested convention.

Thanks!

**Comment by ecordell** [comment]
What version of docker are you using? It sounds like you have an older version that's not supporting the image labels (`as builder`)

The `FROM golang:1.10` refers to a docker image, not your local golang installation.

**Comment by benjaminapetersen** [comment]
Aha, that makes sense.  I probably should have inferred from the caps `FROM`...

So I'm running Docker, on Mac, @ 1.13.1.
This is intentional for other reasons on my system.  Not sure if there is a handy workaround...

**Comment by benjaminapetersen** [comment]
I can give it another shot in a VM w/a current Docker & see where we land.

**Comment by benjaminapetersen** [comment]
@ecordell Updating to the latest docker (1.18).

I can now run the build (though it fails) with:

```bash
docker build -t quay.io/benjaminapetersen/myapp-operator --build-arg HELM_CHART=examples/myapp/ --build-arg API_VERSION=app.benjaminapetersen.me/v1alpha1 --build-arg KIND= MyApp .
```
With the output of:

```bash 
Sending build context to Docker daemon  337.2MB
Step 1/20 : FROM golang:1.10 as builder
 ---> d0e7a411e3da
Step 2/20 : ARG HELM_CHART
 ---> Using cache
 ---> a601d3a18a00
Step 3/20 : ARG API_VERSION
...
... redacted ....
...
Step 18/20 : RUN mkdir /chart   && tar -xzf /chart.tgz --strip-components=1 -C /chart   && rm /chart.tgz
 ---> Running in 6a1f2e491c55
tar: invalid magic
tar: short read
The command '/bin/sh -c mkdir /chart   && tar -xzf /chart.tgz --strip-components=1 -C /chart   && rm /chart.tgz' returned a non-zero code: 1
```
Closer!

**Comment by benjaminapetersen** [comment]
"invalid magic" is kind of a fantastic output, btw. 😄

**Comment by alecmerdler** [comment]
Yes, this is because you are trying to use a local Chart directory instead of a remotely hosted compressed directory. Docker build has [slightly inconsistent behavior of the `ADD` command](https://docs.docker.com/engine/reference/builder/#add).

> If `<src>` is a local tar archive in a recognized compression format (identity, gzip, bzip2 or xz) then it is unpacked as a directory. Resources from remote URLs are not decompressed. 

The reason the `Dockerfile` was written this way was to let users quickly create an Operator using a pre-made, remotely hosted Helm Chart. Obviously we need more logic here to determine if you are trying to use a local, uncompressed Chart directory.

Do you mind filing a separate issue or making a PR to handle this case?

**Comment by benjaminapetersen** [comment]
Issue coming right up!

**Comment by alecmerdler** [comment]
Fixed in #28.
